File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int solution(string my_string) 
    {
        string numStr = "";
        List<string> numList = new List<string>();
        bool isInt = false;
        for (int i = 0; i < my_string.Length; i++)
        {
            // 현재 문자가 숫자인 경우
            if (Char.IsDigit(my_string[i]))
            {
                numStr += my_string[i];
            } // 마지막 index에 숫자가 담긴다면 문자열에 붙지만 리스트에 추가는 안되므로 생각을 잘해야함!
            else
            {
                // 현재 문자가 숫자가 아니고, numStr에 숫자가 담겨있다면
                if (numStr.Length != 0)
                {
                    numList.Add(numStr);
                    numStr = ""; // numStr 초기화
                }
            }
        }

        // 문자열의 마지막 부분에 숫자가 남아 있을 수 있음
        if (numStr.Length != 0)
        {
            numList.Add(numStr);
        }
        
        int numSum = 0;
        for(int i=0; i<numList.Count; i++)
        {
            numSum += Int32.Parse(numList[i]);
        }
        return numSum;
    }
}

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string my_string) 
{
    vector<int> num;
    string temp = "";
    for(char &c : my_string)
    {
        if(isdigit(c))
        {
            temp += c;
        }
        if(!isdigit(c) && !temp.empty())
        {
            num.push_back(stoi(temp));
            temp = "";
        }
    }
    if(!temp.empty()) { num.push_back(stoi(temp)); }
    
    int sum = 0;
    for(int &i : num)
    {
        sum += i;
    }
    return sum;
}